tests/ResourceSyncingPolymorphicTest.php

```diff
@@ -0,0 +1,334 @@
+        'role' => 'commenter',
+    ]);
+
+    tenancy()->end();
+
+    // Assert User resource is synced
+    $centralUser = CentralUserForPolymorphic::first()->only(['name', 'email', 'password', 'role']);
+    $resourceUser = $resourceUser->only(['name', 'email', 'password', 'role']);
+    expect($resourceUser)->toBe($centralUser);
+
+    $tenant2 = ResourceTenantForPolymorphic::create(['id' => 't2']);
+    migrateCompaniesTableForTenants();
+
+    tenancy()->initialize($tenant2);
+
+    $resourceCompany = ResourceCompanyForPolymorphic::create([
+        'global_id' => 'acme',
+        'name' => 'tenant comp',
+        'email' => 'company@localhost',
+    ]);
+
+    tenancy()->end();
+
+    // Assert Company resource is synced
+    $centralCompany = CentralCompanyForPolymorphic::first()->only(['name', 'email']);
+    $resourceCompany = $resourceCompany->only(['name', 'email']);
+    expect($resourceCompany)->toBe($centralCompany);
+});
+
+function migrateCompaniesTableForTenants(): void
+{
+    pest()->artisan('tenants:migrate', [
+        '--path' => __DIR__ . '/Etc/synced_resource_migrations/companies',
+        '--realpath' => true,
+    ])->assertExitCode(0);
+}
+
+// todo better names for resource syncing setup here
+
+class ResourceTenantForPolymorphic extends Tenant
+{
+    public function users(): MorphToMany
+    {
+        return $this->morphedByMany(CentralUserForPolymorphic::class, 'tenant_resources', 'tenant_resources', 'tenant_id', 'resource_global_id', 'id', 'global_id')
+            ->using(TenantPivot::class);
+    }
+
+    public function companies(): MorphToMany
+    {
+        return $this->morphedByMany(CentralCompanyForPolymorphic::class, 'tenant_resources', 'tenant_resources', 'tenant_id', 'resource_global_id', 'id', 'global_id')
+            ->using(TenantPivot::class);
+    }
+}
+
+class CentralUserForPolymorphic extends Model implements SyncMaster
+{
+    use ResourceSyncing, CentralConnection;
+
+    protected $guarded = [];
+
+    public $timestamps = false;
+
+    public $table = 'users';
+
+    // override method to provide different tenant
+    public function getResourceTenantModelName(): string
+    {
+        return ResourceTenantForPolymorphic::class;
+    }
+
+    public function getTenantModelName(): string
+    {
+        return ResourceUserForPolymorphic::class;
+    }
+
+    public function getGlobalIdentifierKey(): string|int
+    {
+        return $this->getAttribute($this->getGlobalIdentifierKeyName());
+    }
+
+    public function getGlobalIdentifierKeyName(): string
+    {
+        return 'global_id';
+    }
+
+    public function getCentralModelName(): string
+    {
+        return static::class;
+    }
+
+    public function getSyncedAttributeNames(): array
+    {
+        return [
+            'global_id',
+            'name',
+            'password',
+            'email',
+        ];
+    }
+}
+
+class ResourceUserForPolymorphic extends Model implements Syncable
```

Comment:
Refer back to our conversation about `ResourceUser` and properly naming things in resource syncing tests.